piker/brokers/ib.py

```diff
@@ -1183,112 +1201,217 @@ def pack_position(pos: Position) -> Dict[str, Any]:
     else:
         symbol = con.symbol
 
-    return {
-        'broker': 'ib',
-        'account': pos.account,
-        'symbol': symbol,
-        'currency': con.currency,
-        'size': float(pos.position),
-        'avg_price': float(pos.avgCost) / float(con.multiplier or 1.0),
-    }
+    return BrokerdPosition(
+        broker='ib',
+        account=pos.account,
+        symbol=symbol,
+        currency=con.currency,
+        size=float(pos.position),
+        avg_price=float(pos.avgCost) / float(con.multiplier or 1.0),
+    )
 
 
-@tractor.msg.pub(
-    send_on_connect={'local_trades': 'start'}
-)
-async def stream_trades(
+async def handle_order_requests(
+
+    ems_order_stream: tractor.MsgStream,
+
+) -> None:
+
+    # request_msg: dict
+    async for request_msg in ems_order_stream:
+        log.info(f'Received order request {request_msg}')
+
+        action = request_msg['action']
+
+        if action in {'buy', 'sell'}:
+            # validate
+            order = BrokerdOrder(**request_msg)
+
+            # call our client api to submit the order
+            reqid = await _trio_run_client_method(
+
+                method='submit_limit',
+                oid=order.oid,
+                symbol=order.symbol,
+                price=order.price,
+                action=order.action,
+                size=order.size,
+
+                # XXX: by default 0 tells ``ib_insync`` methods that
+                # there is no existing order so ask the client to create
+                # a new one (which it seems to do by allocating an int
+                # counter - collision prone..)
+                reqid=order.reqid,
+            )
+
+            # deliver ack that order has been submitted to broker routing
+            await ems_order_stream.send(
+                BrokerdOrderAck(
+                    # ems order request id
+                    oid=order.oid,
+                    # broker specific request id
+                    reqid=reqid,
+                    time_ns=time.time_ns(),
+                ).dict()
```

Comment:
Once we get some better integration for `msgpack-python` and `pydantic.BaseModel` (and/or `msgspec.Struct`) with `tractor`, we can just send the objects directly instead of coercing to dicts.